### ISSUE TITLE
Ignore lines of code

### ISSUE THREAD

**Issue by Glavin001** [Unibeautify/unibeautify#39]
- [x] `ignore-next-line` - Ignore changes for next line
- [ ] `ignore-line` - Ignore changes for current line
- [ ] `disable-beautifier` and `enable-beautifier` - Temporarily disable any changes by beautifier(s) until re-enabled

---

See:
- https://prettier.io/docs/en/ignore.html
- https://palantir.github.io/tslint/usage/rule-flags/
- https://eslint.org/docs/user-guide/configuring.html#disabling-rules-with-inline-comments
- https://github.com/beautify-web/js-beautify/blob/d0924534cae2d024bfaefb4ab64a67d98c79e812/README.md#directives-to-ignore-or-preserve-sections-javascript-only

**Comment by Glavin001** [comment]
Should first check if original code contains possible keywords: https://stackoverflow.com/a/5296314/2578205

**Comment by Glavin001** [comment]
Use https://yarnpkg.com/en/package/diff to determine the differences and then check for comments which tell Unibeautify to ignore changes for these lines.

**Comment by szeck87** [comment]
1. Check original code for keywords (either inside or outside of comments)
2. Beautify the code
3. If any are found, use the diff tool to determine the differences between the original and beautified contents
4. Read code to really check if there are comments that tell the beautifier to ignore parts of the code, and replace beautified content with original content

Is that right?

**Comment by Glavin001** [comment]
Exactly! This way we can support it for all beautifiers. At least, that's the idea. Will be testing this theory today.

**Comment by Glavin001** [comment]
Published to v0.9.0

**Comment by Glavin001** [comment]
`ignore-next-line` is not working! 🔥

**Comment by Glavin001** [comment]
A few related bugs:
- https://github.com/kpdecker/jsdiff/issues/157
- https://github.com/kpdecker/jsdiff/issues/209
- https://github.com/kpdecker/jsdiff/issues/94

**Comment by Glavin001** [comment]
Published to v0.9.1

**Comment by Glavin001** [comment]
Working great now!

**Comment by Glavin001** [comment]
@lassik this feature is a neat experiment, however, I do agree it's a little too ambitious to expect it to function without errors and would distract from the other valuable goals of Unibeautify. I think your comment sums it up nicely:

> If we have a policy of doing nothing except calling external formatters, that policy is simple for users to understand, and any bugs in formatting can simply be directed to a particular beautifier's bug tracker.

👍.

Fortunately, I don't think anyone is using this feature too heavily. I'll investigate removing it. Pull Requests welcome.
